packages/babel-traverse/test/evaluation.js

```diff
@@ -37,4 +37,11 @@ suite("evaluation", function () {
       false
     );
   });
+
+  test("should not be confident when var is redeclared in the same scope", function () {
+    assert.strictEqual(
+      getPath("var x = 2; var y = x + 2; { var x = 3 }").get("body.1.declarations.0.init").evaluate().confident,
```

Comment:
Looks good to fix the bug, but as another TODO `y` be able to resolve to `4` right?